### ISSUE TITLE
Idea: allow for customizable zlobby data directory

### ISSUE THREAD

**Issue by verybadsoldier** [marciomarim/zlobby#5]
In order to use zlobby for multiple games in parallel (e.g. BA and BAR) the user has to store different user credentials. Also if you want the games to be isolated they should use different spring directories also.

At the moment Zlobby seems to store its config in `%appdata%\zlobby` with a single user credential and with a single spring directory.

If it would be possible for the user to pass a custom data directory then you could have completeley different zlobby configurations for BA and BAR.
E.g. one in `%appdata%\zlobby_BA` and one in `%appdata%\zlobby_BAR`. Then you could start zlobby with the directory you want to use and you could have isolated configurations for BA and BAR.

Maybe this would be easier to implement than handling per-game configurations inside of zlobby? Just an idea...

**Comment by marciomarim** [comment]
I like the idea of isolation and I already have the solution for different credentials depending on the host.

But for game isolation I don't want to have doubled maps and games folder. Are we talking about one springsettings.cfg per server? 

And if so, do you know if we can pass a specific springsettings.cfg when we launch spring.exe? That would solve the problem maybe.

**Comment by verybadsoldier** [comment]
Yes, I think springsettings.cfg is the main issue. But I don't know well enough how it is loaded (it also loads engine-specific settings afaik) nor how to force a specific setting file.

But in general there could be also other conflicting datas when having multiple games in same spring directory. What comes to mind (maybe there is more):
* Installed user widgets - game's widget handler chooses freely where to load widgets from in the VFS. So I think you can't really prevent different games from loading each other's wigets if the widget handler is implemented that way
* Widget configuration - should be fine as long as games are using unique identifies which we can probably assume to be the case
* Custom widget data - AFAIK widgets can read/write arbitrary files at least in some parts of the VFS. Also not possible to prevent interaction between games

But probably users can live with that. But I think the most clean isolation is by using different spring directories.

Imo the only real data that is desirable to be shared between games are the maps as they are immutable. In case of BA/BAR it might even be a minor problem since they don't share many common maps? I think BAR exclusively is played on Behe's BAR maps while BA rarely is played on them.

But I don't really have deeper knowledge how spring loads is settings or how game or widget data is handled in detail. So it might be just wrong in the end :)
